upstairs/src/lib.rs

```diff
@@ -3437,6 +3447,16 @@ pub struct Upstairs {
      * Upstairs stats.
      */
     stats: UpStatOuter,
+
+    /*
+     * Does this Upstairs throw random errors?
+     */
+    lossy: bool,
+
+    /*
+     * Operate in read-only mode
+     */
+    read_only: bool,
```

Comment:
If read_only is part of the `Upstairs` struct now, how do things work with volumes that have a ReadOnly part and a RW part?  Does the Volume layer handle that, and the upstairs that serves a snapshot will be a different upstairs than what serves the RW part?  I'm just a little confused still, sorry.